src/asm/lexer.cpp

```diff
@@ -1849,7 +1849,18 @@ static Token yylex_NORMAL() {
 					}
 				}
 
-				if (token.type == T_(ID) && (lexerState->atLineStart || peek() == ':'))
+				// This is a "lexer hack"! We need it to distinguish between label definitions
+				// (which start with `LABEL`) and macro invocations (which start with `ID`).
+				//
+				// If we had one `IDENTIFIER` token, the parser would need to perform "lookahead"
+				// to determine which rule applies. An "IDENTIFIER COLON" rule would work for
+				// labels, but any macro rule would need to enter "raw" mode right after parsing
+				// the `IDENTIFIER`, which would prevent lookahead from recognizing a `COLON`.
```

Comment:
That is incorrect. We could, in theory, concat the token's representation back into the macro argument after failing to match the token following the `IDENTIFIER`.

Problem is, macro arguments can contain syntax that's outright *invalid* in the context of the macro invocation, and so we *cannot* afford to lex a token at all! (Remember that C's preproc requires that all token operations still yield valid pp-tokens? That's why.)

Reply:
Reworded.

Reply:
Perfect, thank you!